### PR TITLE
Cleanup session code, docs fixes for Shopify

### PR DESCRIPTION
**Pull request by davidmuzi**
- consolidates access_token storage in all services
- makes shop_domain public
- docs fixes

### CODE REVIEW

File: Sources/Imperial/Services/GitHub/GitHubRouter.swift [hunk 1]
```diff
@@ -46,8 +46,8 @@ public class GitHubRouter: FederatedServiceRouter {
     public func callback(_ request: Request)throws -> Future<Response> {
         return try self.fetchToken(from: request).flatMap(to: ResponseEncodable.self) { accessToken in
             let session = try request.session()
-            
-            session["access_token"] = accessToken
+
```
Comment (robinsenior) [inline Sources/Imperial/Services/GitHub/GitHubRouter.swift]: 👀

File: Sources/Imperial/Helpers/Sessions+Imperial.swift [hunk 2]
```diff
@@ -15,17 +15,29 @@ extension Request {
 }
 
 extension Session {
-    
+	
+	enum Keys {
```
Comment (calebkleveter) [inline Sources/Imperial/Helpers/Sessions+Imperial.swift]: Nitpick: The indentation whitespace for this enum should match the indentation for the rest of the extensions methods. Also, the whitespace above it should be removed to it matches the rest of the extensions and types in the project.

File: Sources/Imperial/Helpers/Sessions+Imperial.swift [hunk 3]
```diff
@@ -15,17 +15,29 @@ extension Request {
...
             throw Abort(.unauthorized, reason: "User currently not authenticated")
         }
         return token
     }
+	
+	
+	/// Sets the access token on the session.
+	///
+	/// - Parameter token: the access token to store on the session
+	public func setAccessToken(_ token: String) {
```
Comment (calebkleveter) [inline Sources/Imperial/Helpers/Sessions+Imperial.swift]: Fix indentation like the enum above.

**Comment by calebkleveter** [comment]
@davidmuzi Are you ready for this to be merged, or do you want to make more changes first?

**Comment by davidmuzi** [comment]
Good to merge!

I was running through starting a new app from scratch and was running into this fatal error: 
```swift
public var container: Container {
        guard let c = _container else {
            fatalError("If you encounter this error, you are holding on to a client after de-initializing your Application. This is usually a bad idea.")
        }
        return c
    }
```

It looks like there where 2 `request` objects in `public func fetchToken(from request: Request) throws -> EventLoopFuture<String>` that where in scope, and the wrong one was being used.  Oddly, the app I created along side the work in Imperial works fine?  Either way, it obviously makes sense that I renamed the variables to not cause this confusion.

**Pull request merged**